sqlglot/generator.py

```diff
@@ -959,9 +962,16 @@ def fetch_sql(self, expression: exp.Fetch) -> str:
         return f"{self.seg('FETCH')}{direction}{count} ROWS {with_ties_or_only}"
 
     def filter_sql(self, expression: exp.Filter) -> str:
-        this = self.sql(expression, "this")
-        where = self.sql(expression, "expression").strip()
-        return f"{this} FILTER({where})"
+        if self.AGGREGATE_FILTER_SUPPORTED:
+            this = self.sql(expression, "this")
+            where = self.sql(expression, "expression").strip()
+            return f"{this} FILTER({where})"
+
+        agg = expression.this.copy()
+        agg_arg = agg.this
+        cond = expression.expression.this
+        agg_arg.replace(exp.If(this=cond.copy(), true=agg_arg.copy(), false=exp.null()))
```

Comment:
the false is unnecessary  it’s implied null

Reply:
fixed this in https://github.com/tobymao/sqlglot/pull/2243/commits/d5ad990a93d438a5cf748eb754e6e4daaefd326e